rpc/handlers.go

```diff
@@ -32,18 +37,26 @@ const (
 )
 
 type Handler struct {
-	bcReader     blockchain.Reader
-	synchronizer *sync.Synchronizer
-	network      utils.Network
-	log          utils.Logger
+	bcReader      blockchain.Reader
+	synchronizer  *sync.Synchronizer
+	network       utils.Network
+	gatewayClient gateway.Gateway
+	log           utils.Logger
 }
 
-func New(bcReader blockchain.Reader, synchronizer *sync.Synchronizer, n utils.Network, logger utils.Logger) *Handler {
+func New(
+	bcReader blockchain.Reader,
+	synchronizer *sync.Synchronizer,
+	n utils.Network,
+	gatewayClient gateway.Gateway,
+	logger utils.Logger,
+) *Handler {
```

Comment:
Let's keep them on a single line. If the linter complains then do the following:

```go
func New(cReader blockchain.Reader, synchronizer *sync.Synchronizer, n utils.Network, gatewayClient gateway.Gateway,
 logger utils.Logger) *Handler {
```

Only move the minimum number of args required to the next line to satisfy `lll`.